Assets/Project/Scripts/GamePlayScene/Bottle/IEnterTileHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿using SpriteGlow;
+using UnityEngine;
+
+namespace Project.Scripts.GamePlayScene.Bottle
+{
+    public interface IEnterTileHandler
+    {
+        void OnEnterTile(GameObject tile);
+    }
+
+    internal class NormalEnterTileHandler : IEnterTileHandler
+    {
+        private readonly AbstractBottleController _bottle;
+        private readonly IBottleSuccessHandler _successHandler;
+
+        internal NormalEnterTileHandler(AbstractBottleController bottle, IBottleSuccessHandler successHandler)
+        {
+            _bottle = bottle;
+            _successHandler = successHandler;
+        }
+
+        public void OnEnterTile(GameObject tile)
+        {
+            if (_successHandler.IsSuccess()) {
+                // 最終タイルにいるかどうかで，光らせるかを決める
+                _bottle.GetComponent<SpriteGlowEffect>().enabled = true;
+
+                _successHandler.DoWhenSuccess();
```

Comment:
```suggestion
                _bottle.DoWhenSuccess();
```
という気持ちの実装に出来ませんか？
`IEnterTileHandler`が`IBottleSuccessHandler( = _successHandler)`を必要とするのに違和感があります。

Reply:
`IEnterTileHandler` が `IBottleSuccessHandler` を必要とするのではなくて、 `NormalEnterTileHandler` が `IBottleSuccessHandler` を必要とすることを考えて欲しいです。
なぜなら、今の仕様は「タイルに入る時は成功判定する」からです。

Reply:
なるほど違和感の原因を理解した。
`NormalEnterTileHandler`が成功判定の実装を必要とするならコンストラクタとかで`null`判定してほしい。

もしくは(こっちの方が自分は納得できるけど、意に反するかもしれない)、成功判定だけは`AbstractBottleController`が持つべきじゃなくて、成功判定を持つクラスが`implements`すべきだと思う。
`移動する`、`衝突する`はイベントの発生と処理が同時だけど、`成功する`は成功する処理はあっても成功する、というイベントは発生しないから他2つとは異なると思う。
(まだ成功判定以外にこういう機能が無いから現時点では今の方針でも納得できるけど、どう思うかは聞きたい。)
(ちょっとニュアンスが言葉にしづらいなあ。。。)

Reply:
ごめん、ちょっとわからなかったから、どうしてほしいかをサンプルコードを書いていただけると助かる